### DISCUSSION TITLE
Dockerise ASP.NET Core app with Alpine image

### DISCUSSION THREAD

**Discussion by javiercn** [dotnet/dotnet-docker#4852] [Q&A]
---
**Issue moved from dotnet/aspnetcore#50399**
- Please respond to @lonix1.

---

_From @lonix1 on Tuesday, August 29, 2023 3:46:57 PM_

I originally asked on [SO](https://stackoverflow.com/q/77000895/9971404) without resolution. I assume this is not working as intended.

---

I want my dockerised ASP.NET Core 7 app based on [Alpine](https://hub.docker.com/_/alpine) (not [ASP.NET Core Runtime](https://hub.docker.com/_/microsoft-dotnet-aspnet)).

`MyApp.csproj` contains:
```xml
<PublishSingleFile>true</PublishSingleFile>
<RuntimeIdentifier>linux-musl-x64</RuntimeIdentifier>
<IncludeNativeLibrariesForSelfExtract>true</IncludeNativeLibrariesForSelfExtract>  <!-- tried with and without this -->
```

`Dockerfile` contains:
```docker
FROM alpine
```

But when I start a container, I get:
```
Error relocating /dist/MyApp: __cxa_pure_virtual: symbol not found
Error relocating /dist/MyApp: __cxa_pure_virtual: symbol not found
Error relocating /dist/MyApp: __cxa_pure_virtual: symbol not found
Error relocating /dist/MyApp: _ZTVN10__cxxabiv121__vmi_class_type_infoE: symbol not found
Error relocating /dist/MyApp: _ZTVN10__cxxabiv121__vmi_class_type_infoE: symbol not found
Error relocating /dist/MyApp: _ZTVN10__cxxabiv121__vmi_class_type_infoE: symbol not found
```

It works with the normal image (i.e. `FROM mcr.microsoft.com/dotnet/aspnet`).

The [docs](https://learn.microsoft.com/en-us/dotnet/core/deploying/single-file/overview) imply that with the correct RID (`linux-musl-x64`), this should work. But obviously the generated binary is missing native dependencies.

**Comment by javiercn** [comment]
---
**Issue moved from dotnet/aspnetcore#50399**
- Please respond to @lonix1.

---

_From @lonix1 on Wednesday, August 30, 2023 1:26:31 AM_

### UPDATE:

The answers in that [SO thread](https://stackoverflow.com/q/77000895/9971404) explain that the `runtime-deps` image is needed for various alpine dependencies.

*So this is working properly, but the documentation needs some polish.*

Somewhere in [[1]](https://learn.microsoft.com/en-us/dotnet/core/deploying/single-file/overview) or [[2]](https://learn.microsoft.com/en-us/dotnet/core/rid-catalog) it should mention that the self-contained single-file app cannot be run in `alpine` directly, but can be run in `runtime-deps`.

Could you possibly move this to the docs repo?

**Comment by javiercn** [comment]
---
**Issue moved from dotnet/aspnetcore#50399**
- Please respond to @javiercn.

---

_From @javiercn on Wednesday, August 30, 2023 10:24:32 AM_

@lonix1 thanks for contacting us.

This is more likely a question for the dotnet/docker repository since they centrally manage all images.

@richlander thoughts?

**Comment by mthalman** [comment]
This was already answered here: https://stackoverflow.com/questions/77000895/dockerise-asp-net-core-app-with-alpine-image
